src/plugins/justwatch/hub.ts

```diff
@@ -0,0 +1,179 @@
+import * as plexTypes from '../../plex/types';
+import {
+	PseuplexFeedHub,
+	PseuplexFeedHubChunk,
+	PseuplexFeedHubOptions,
+	PseuplexRequestContext,
+	PseuplexMetadataTransformOptions
+} from '../../pseuplex';
+import { LoadableListItemNode } from '../../fetching/LoadableListFragment';
+import { JustWatchHubConfig, JustWatchTitle, JustWatchLanguage, JustWatchCountry, JustWatchSortBy } from './types';
+import { JustWatchMetadataProvider } from './metadata';
+import { getPopularTitles } from './api';
+
+export type JustWatchHubOptions = PseuplexFeedHubOptions & {
+	config: JustWatchHubConfig;
+	justWatchMetadataProvider?: JustWatchMetadataProvider;
+};
+
+type JustWatchHubChunk = PseuplexFeedHubChunk<JustWatchTitle, void, string>;
+
+export class JustWatchHub extends PseuplexFeedHub<JustWatchTitle, void, string, JustWatchHubOptions> {
```

Comment:
Since there are multiple methods that call the graphql api, it would be better if you passed the fetcher similar to [here](https://github.com/lufinkey/pseuplex/blob/16e18dc66eff59a2c8451b5c4b336eb9a79e733f/src/plugins/letterboxd/filmshub.ts#L25), that way you can define a bunch of different hubs that use the same hub object, like [here](https://github.com/lufinkey/pseuplex/blob/dev/src/plugins/letterboxd/hubs.ts)

Reply:
right now in that file, they're all using their own hub classes because they use different apis, but since there are other methods in the letterboxd api class that return similar objects, i can reuse them when I add support for those